core/src/main/java/org/apache/calcite/tools/RelBuilder.java

```diff
@@ -2503,7 +2503,7 @@ private RelBuilder pruneAggregateInputFieldsAndDeduplicateAggCalls(
           builder.add(project.getRowType().getFieldList().get(i));
         }
         r =
-            project.copy(cluster.traitSet(), project.getInput(), newProjects,
+            project.copy(project.getTraitSet(), project.getInput(), newProjects,
```

Comment:
Can we always propagate all traits? Since this happens when the fields of an existing project are pruned down, couldn't a Collation or Distribution trait end up being set incorrectly?

Reply:
I guess to be safe we should just propagate the convention.

Reply:
I've changed the code to only propagate the convention.

Reply:
I think there might be a slightly better way using Mapping, the column pruning is effectively a Mapping anyway, so if we were to make it actually a Mapping there's a method that will transform the TraitSet according to that mapping, I believe including Collation and Distribution

Reply:
This is what I was thinking of
`public RelTraitSet apply(Mappings.TargetMapping mapping)`

Reply:
Done. The post-pruning RelTraitSet is now calculated using RelTraitSet.apply(Mapping) from the original project's RelTraitSet.